gateway-ha/src/test/java/io/trino/gateway/ha/handler/TestQueryIdCachingProxyHandler.java

```diff
@@ -34,32 +45,289 @@ void testExtractQueryIdFromUrl()
+                extractQueryIdIfPresent(
+                        prepareMockRequestWithBody("""
+                                CALL
+                                KILL_QUERY
+                                (
+                                -- this is a comment
+                                '20200416_160256_03078_6b4yt' --this is a trailing comment
+                                ,
+                                /*
+                                this is
+                                a multiline comment
+                                */
+                                'If he dies, he dies
+                                ')
+                                """, "system", "runtime"),
+                        ImmutableList.of(),
+                        requestAnalyzerClientsUseV2Format,
+                        requestAnalyzerMaxBodySize))
+                .hasValue("20200416_160256_03078_6b4yt");
+
+        assertThatThrownBy(() -> extractQueryIdIfPresent(
+                prepareMockRequestWithBody("""
+                                CALL KILL_QUERY (lower('20200416_160256_03078_6b4yt'), 'If he dies, he dies')
+                                """, "system", "runtime"),
+                ImmutableList.of(),
+                requestAnalyzerClientsUseV2Format,
+                requestAnalyzerMaxBodySize)).isInstanceOf(IllegalArgumentException.class);
+
+        assertThat(
+                extractQueryIdIfPresent(
+                        prepareMockRequestWithBody("CALL notsystem.runtime.kill_query(query_id => '20200416_160256_03078_6b4yt', message => 'If he dies, he dies')"),
+                        ImmutableList.of(), requestAnalyzerClientsUseV2Format, requestAnalyzerMaxBodySize))
+                .isEmpty();
+
+        assertThat(
+                extractQueryIdIfPresent(
+                        prepareMockRequestWithBody("CALL runtime.kill_query(query_id => '20200416_160256_03078_6b4yt', message => 'If he dies, he dies')", "notsystem"),
+                        ImmutableList.of(), requestAnalyzerClientsUseV2Format, requestAnalyzerMaxBodySize))
+                .isEmpty();
+
+        assertThat(
+                extractQueryIdIfPresent(
+                        prepareMockRequestWithBody("CALL notruntime.kill_query(query_id => '20200416_160256_03078_6b4yt', message => 'If he dies, he dies')", "system"),
+                        ImmutableList.of(), requestAnalyzerClientsUseV2Format, requestAnalyzerMaxBodySize))
+                .isEmpty();
+
+        assertThat(
+                extractQueryIdIfPresent(
+                        prepareMockRequestWithBody("CALL kill_query(query_id => '20200416_160256_03078_6b4yt', message => 'If he dies, he dies')", "system", "notruntime"),
+                        ImmutableList.of(), requestAnalyzerClientsUseV2Format, requestAnalyzerMaxBodySize))
+                .isEmpty();
+    }
+
+    private static HttpServletRequest prepareMockRequestWithBody(String query, String defaultCatalog)
```

Comment:
I would rename it to `request`. Longer helper method name reduces readability of each assertion.